### DISCUSSION TITLE
Use a VPN tunnel with one phy connection to anonymize data transfer

### DISCUSSION THREAD

**Discussion by sorryusernameisalreadytaken** [extremecoders-re/go-dispatch-proxy#31] [Q&A]
First of all: Nice that this project exists!

My internet provider throttles every data connection to 200 Mbps. With multiple data connections, I always get my full 500 Mbps.
Just about every speedtest for internet speeds uses at least 3 persistent data connections to make a test. Unfortunately, this is also recommended by the BEREC Net Neutrality Regulatory Assessment Methodology.

So I would like to use go-dispatch-proxy not to load-balance multiple physical connections, but one pyhysical connection with multiple simultaneous connections. I would like to do this to remove the throttling.

Secondly, I always use a wireguard connection to a commercial provider to tunnel and anonymize my internet packets through them.

Is this combination with go-dispatch-proxy possible in a linux (ubuntu) environment?

**Comment by extremecoders-re** [comment]
Yes it should be possible.
You can create multiple wireguard interfaces like wg0, wg1 & wg2 with different subnets. Then you should be able to use go-dispatch-proxy on each of these three interfaces.

**Reply by sorryusernameisalreadytaken** [reply]
Would you have thought that it is possible to make this work with a Docker Compose based environment?

I place here my other discussion in gleutun project, I think is a good combination of this question:
https://github.com/qdm12/gluetun/discussions/1455
